plugins/cros-ec/fu-cros-ec-usb-device.c

```diff
@@ -481,8 +482,17 @@ fu_cros_ec_usb_device_transfer_block (FuDevice *device, gpointer user_data,
 					    sizeof(struct update_frame_header),
 					    NULL,
 					    0, FALSE,
-					    NULL, error))
+					    NULL, error)) {
+		g_autoptr(GError) error_local = NULL;
+
+		/* flush all data from endpoint to recover in case of error */
+		if (!fu_device_retry (device, fu_cros_ec_usb_device_flush,
+				      SETUP_RETRY_CNT, NULL, &error_local)) {
+			g_debug ("failed to flush to idle");
```

Comment:
if you're not actually showing the message from `error_local` here, I would just send `NULL` for `fu_device_retry`

Reply:
Also, since this is (essentially) the same code 3 times, does this want a static function to call into instead?

Reply:
Done. Thanks. This replaces 5 instances.